tests/utils/managed_process.py

```
@@ -0,0 +1,242 @@
+            self._sed_proc = subprocess.Popen(
+                ["sed", "-u", f"s/^/[{self._command_name.upper()}] /"],
+                stdin=self.proc.stdout,
+                stdout=subprocess.PIPE,
+            )
+
+            self._tee_proc = subprocess.Popen(
+                ["tee", self._log_path], stdin=self._sed_proc.stdout
+            )
+
+        else:
+            with open(self._log_path, "w") as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)